### PR TITLE
Fix column widths in view stock within inventory

### LINKED ISSUE #2462: Stock page: Column headers jump left-right sometimes
## What went wrong? 😲
On the stock page the headers jump left ot right sometimes when scrolling up and down the list. See the recording here:

https://github.com/msupply-foundation/open-msupply/assets/11517184/620d0dba-08ba-47db-87f6-92ad2cf63c95

## Expected behaviour 🤔
No jumping headers

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Go to Inventory > View stock in the menus
2. Scroll down the list of lines. Headers might jump
3. Click on different headers to order by different columns and repeat 
4. See jumping

## Your environment 🌱

 - Version: 1.5.00
 - Platform:
   - [ ] android (tablet)
   - [X] browser (extra points if you tell us which one) Firefox 
   - [ ] desktop (windows)
   - [ ] desktop (macOS)
   - [ ] server (windows)

### PR DESCRIPTION
**Pull request by fergie-nz**
Fixes #2462

# 👩🏻‍💻 What does this PR do? 
Fixes the column widths of some columns of stock listview.

# 🧪 How has/should this change been tested? 
- [ ] view stock in Inventory
- [ ] See that columns don't jump when either name or batch names extend beyond the normal width of the columns. See issue #2462 for an example of column size jumping.

## 💌 Any notes for the reviewer?
There might be some other cases where fixing column widths is necessary to prevent jumping, I'm not sure.
I kept this change to a minimum by only changing the obvious jumping columns in the stock list view.

### CODE REVIEW

**Review by mark-prins — CHANGES_REQUESTED** [review]
I may be old, but I've still got it!

jumpy-stock

by which I mean, looks like we need to fix the widths of all columns.

an alternative would be to disable the `ViewportList` in `DataTable` - that's the cause of this. Though that was put in because the long tables were super slow to render on android, so we should probably retain it.

**Comment by fergie-nz** [comment]
What do you think of this?

I feel a bit strange fixing the widths when I think about different devices using the table, and with the reliance on tooltips which won't work on tablet.

But perhaps still better?

I've also added fixing widths for plugin columns to avoid duplication / reliance on specifying fixed widths there.

**Review by mark-prins — CHANGES_REQUESTED** [review]
I'm also a little reluctant to fix the widths in this way.. but it does fix the problem.
On reflection, I think best not to force the plugins though - as it is the plugin will need to specify a width. Why not just require them to set a minWidth and maxWidth?

What about making the column creation easier by [doing this](https://github.com/msupply-foundation/open-msupply/compare/2462-fix-jumping-stock-columns...2462-fix-jumping-stock-columns(column-width))?

File: client/packages/system/src/Stock/ListView/ListView.tsx [hunk 1]
```diff
@@ -87,105 +87,92 @@ const StockListComponent: FC = () => {
...
       key: 'supplierName',
       label: 'label.supplier',
       accessor: ({ rowData }) =>
         rowData.supplierName ? rowData.supplierName : t('message.no-supplier'),
+      Cell: TooltipTextCell,
+      maxWidth: 190,
+      minWidth: 190,
     },
-    ...pluginColumns,
+    ...pluginColumns.map(column =>
```
[the remark this answers is not in the document]
Comment (fergie-nz) [reply client/packages/system/src/Stock/ListView/ListView.tsx:167]: Perfect! Sounds good.

**Review by mark-prins — APPROVED** [review]

**Pull request merged**